### ISSUE TITLE
Release 1.0.4 wheel still installs "futures" in Python 3

### ISSUE THREAD

**Issue by lmazuel** [Azure/azure-cosmosdb-python#31]
Even if this setup.py has a syntax to ignore installing "futures" on Py3 (#23 for reference), the wheel on PyPI still installs it anyway. It's the because the environment where this wheel was *built* is old (older than setuptools 36.2) and the option is ignored silently (old setuptools cannot know it's missing something...).
More detail in this excellent blog
https://hynek.me/articles/conditional-python-dependencies/

There is several possible fixes to this problem:
- Most simple: be sure the build environment is using latest pip/setuptools/wheel always.
- Build and deploy from Travis: https://docs.travis-ci.com/user/deployment/pypi/
- Change the syntax to something allowed in older setuptools. [Example](https://github.com/Azure/msrest-for-python/blob/12fb980c3290daa9c83bacc7e43c3f1455904b8c/setup.py#L55-L58).

It's possible to re-upload a 1.0.4 with the fixed wheel, but probably not worth the effort. But should be addressed with future releases.

**Comment by pklanka** [comment]
Bump this issue. This is causing our Flask modules to fail :(

Any word on when this would get fixed? It has been two months from past release with this bug.

**Comment by srinathnarayanan** [comment]
Hi @lmazuel and @pklanka 
In which environment did you run into this issue?
If you can share a repro, we will update the tools we use to release the package, validate it in that environment and make a quick release.

**Comment by pklanka** [comment]
I dont think environment matters. Just install it in any python 3.6.5 exec env. Futures get collected by setuptools and gets installed. Here is the sample output (edited to represent output better). 

```
$ python
Python 3.6.5 (default, Apr  1 2018, 05:46:30)
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> quit()
ichigo@lankap24 -> ~/aztest
$ python -m pip install azure-cosmosdb-table
Collecting azure-cosmosdb-table
  Using cached https://files.pythonhosted.org/packages/ba/b7/bf3b5891ee87fea0568b5f9d199249c9c3b3208fa3f77256adcda6618558/azure_cosmosdb_table-1.0.4-py2.py3-none-any.whl
Collecting python-dateutil (from azure-cosmosdb-table)
  Using cached https://files.pythonhosted.org/packages/cf/f5/af2b09c957ace60dcfac112b669c45c8c97e32f94aa8b56da4c6d1682825/python_dateutil-2.7.3-py2.py3-none-any.whl
Collecting azure-common>=1.1.5 (from azure-cosmosdb-table)
  Using cached https://files.pythonhosted.org/packages/91/66/6636001c0b6be5423f4bbaffe5febbf9eda39bd1941eb39a73504d8f6f71/azure_common-1.1.14-py2.py3-none-any.whl
Collecting azure-storage-common~=1.1 (from azure-cosmosdb-table)
  Using cached https://files.pythonhosted.org/packages/7b/18/03872177864b7ce4aa0302eae5726e8cb66dbb4f30a3408ed27a6639407f/azure_storage_common-1.3.0-py2.py3-none-any.whl
Collecting azure-cosmosdb-nspkg>=2.0.0 (from azure-cosmosdb-table)
  Using cached https://files.pythonhosted.org/packages/58/0d/1329b47e5386b0acf4e42ada2284851eff60ef3337a87e5b2dfedabbfcb1/azure_cosmosdb_nspkg-2.0.2-py2.py3-none-any.whl
Collecting futures (from azure-cosmosdb-table)
Collecting cryptography (from azure-cosmosdb-table)
  Using cached https://files.pythonhosted.org/packages/59/32/92cade62c645756a83598edf56289e9b19aae5370642a7ce690cd06bc72f/cryptography-2.3.1-cp34-abi3-manylinux1_x86_64.whl
Collecting requests (from azure-cosmosdb-table)
  Using cached https://files.pythonhosted.org/packages/65/47/7e02164a2a3db50ed6d8a6ab1d6d60b69c4c3fdf57a284257925dfc12bda/requests-2.19.1-py2.py3-none-any.whl
Collecting six>=1.5 (from python-dateutil->azure-cosmosdb-table)
  Using cached https://files.pythonhosted.org/packages/67/4b/141a581104b1f6397bfa78ac9d43d8ad29a7ca43ea90a2d863fe3056e86a/six-1.11.0-py2.py3-none-any.whl
Collecting azure-nspkg>=2.0.0 (from azure-common>=1.1.5->azure-cosmosdb-table)
  Using cached https://files.pythonhosted.org/packages/cd/a0/76ca6659ae9afd7567fdbb5b9c85e9c9a0b48238cfcacd92525484408f66/azure_nspkg-2.0.0-py2.py3-none-any.whl
Collecting azure-storage-nspkg>=2.0.0 (from azure-storage-common~=1.1->azure-cosmosdb-table)
  Using cached https://files.pythonhosted.org/packages/b1/e9/0a11f4feefb9c4c2f7e30d4d448e15b32df3adabe370be70aaf4278fbb52/azure_storage_nspkg-3.0.0-py2.py3-none-any.whl
Collecting idna>=2.1 (from cryptography->azure-cosmosdb-table)
  Using cached https://files.pythonhosted.org/packages/4b/2a/0276479a4b3caeb8a8c1af2f8e4355746a97fab05a372e4a2c6a6b876165/idna-2.7-py2.py3-none-any.whl
Collecting asn1crypto>=0.21.0 (from cryptography->azure-cosmosdb-table)
  Using cached https://files.pythonhosted.org/packages/ea/cd/35485615f45f30a510576f1a56d1e0a7ad7bd8ab5ed7cdc600ef7cd06222/asn1crypto-0.24.0-py2.py3-none-any.whl
Collecting cffi!=1.11.3,>=1.7 (from cryptography->azure-cosmosdb-table)
  Using cached https://files.pythonhosted.org/packages/6d/c0/47db8f624f3e4e2f3f27be03a93379d1ba16a1450a7b1aacfa0366e2c0dd/cffi-1.11.5-cp36-cp36m-manylinux1_x86_64.whl
Collecting urllib3<1.24,>=1.21.1 (from requests->azure-cosmosdb-table)
  Using cached https://files.pythonhosted.org/packages/bd/c9/6fdd990019071a4a32a5e7cb78a1d92c53851ef4f56f62a3486e6a7d8ffb/urllib3-1.23-py2.py3-none-any.whl
Collecting certifi>=2017.4.17 (from requests->azure-cosmosdb-table)
  Using cached https://files.pythonhosted.org/packages/16/1f/50d729c104b21c1042aa51560da6141d1cab476ba7015d92b2111c8db841/certifi-2018.8.13-py2.py3-none-any.whl
Collecting chardet<3.1.0,>=3.0.2 (from requests->azure-cosmosdb-table)
  Using cached https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl
Collecting pycparser (from cffi!=1.11.3,>=1.7->cryptography->azure-cosmosdb-table)
Installing collected packages: six, python-dateutil, azure-nspkg, azure-common, azure-storage-nspkg, idna, asn1crypto, pycparser, cffi, cryptography, urllib3, certifi, chardet, requests, azure-storage-common, azure-cosmosdb-nspkg, futures, azure-cosmosdb-table
Successfully installed asn1crypto-0.24.0 azure-common-1.1.14 azure-cosmosdb-nspkg-2.0.2 azure-cosmosdb-table-1.0.4 azure-nspkg-2.0.0 azure-storage-common-1.3.0 azure-storage-nspkg-3.0.0 certifi-2018.8.13 cffi-1.11.5 chardet-3.0.4 cryptography-2.3.1 futures-3.1.1 idna-2.7 pycparser-2.18 python-dateutil-2.7.3 requests-2.19.1 six-1.11.0 urllib3-1.23
```

**Comment by lmazuel** [comment]
Agree with @pkalan , the env does not matter. The metadata embedded inside the wheel are wrong, because the build env of the wheel was old. 
You need to be sure that the release env has at least setuptools 36.2. If you want to test it, installing the resulting wheel in a Python 3 env must NOT install "futures". No exceptions.

**Comment by MartinForReal** [comment]
Any plan to fix this issue? @srinathnarayanan 

I tried to install dependencies from requirement.txt
` 
azure==4.0.0
`

`
root@77ad0a36fd57:/jakiro# python3 --version
Python 3.4.3
root@77ad0a36fd57:/jakiro# pip --version
pip 18.0 from /usr/local/lib/python3.4/dist-packages/pip (python 3.4)
`

but I got 

`
Requirement already satisfied: python-dateutil in /usr/local/lib/python3.4/dist-packages (from azure-cosmosdb-table~=1.0->azure) (2.7.3)
Requirement already satisfied: futures in /usr/local/lib/python3.4/dist-packages (from azure-cosmosdb-table~=1.0->azure) (3.2.0)
futures requires Python '>=2.6, <3' but the running Python is 3.4.3
`

**Comment by srinathnarayanan** [comment]
I apologize for the delay. We have fixed this issue in V 1.0.5. Please reach out to us if this issue still persists

**Comment by lmazuel** [comment]
Thanks @sridharanprasanna !
Side note, you should tag your releases with github:
https://github.com/Azure/azure-cosmosdb-python/releases

Example:
https://github.com/Azure/azure-cli/releases
https://github.com/Azure/azure-storage-python/releases
